lib/checks/aria/required-children.js

```diff
@@ -13,17 +13,17 @@ function owns(node, role, ariaOwned) {
 	}
 
 	selector = selector.join(',');
-
-	return ariaOwned ? (matchesSelector(node, selector) || !!node.querySelector(selector)) :
-		!!node.querySelector(selector);
+	return ariaOwned ? (matchesSelector(node, selector) || !!axe.utils.querySelectorAll(virtualTree, selector)[0]) :
+		!!axe.utils.querySelectorAll(virtualTree, selector)[0];
 }
 
 function ariaOwns(nodes, role) {
 	var index, length;
 
 	for (index = 0, length = nodes.length; index < length; index++) {
 		if (nodes[index] === null) { continue; }
-		if (owns(nodes[index], role, true)) {
+		let virtualTree = axe.utils.getFlattenedTree(nodes[index]);
```

Comment:
I think you should use `axe.utils.getNodeFromTree()` instead of recreating the flattened tree